sympy/physics/mechanics/tests/test_joint.py

```diff
@@ -811,6 +812,135 @@ def test_cylindrical_joint():
+    q0, q1, q2, u0, u1, u2 = dynamicsymbols('q0:3, u0:3')
+    # Test no rotation axis and no planar vectors (default)
+    N, A, P, C = _generate_body()
+    J = PlanarJoint('J', P, C, q0, [q1, q2], u0, [u1, u2],
+                    rotation_axis=None,
+                    planar_vectors=None)
+    assert J.rotation_axis == N.z
+    assert J.planar_vectors == [N.x, N.y]
+    # Test no rotation axis single planar vector parallel to x
+    N, A, P, C = _generate_body()
+    J = PlanarJoint('J', P, C, q0, [q1, q2], u0, [u1, u2],
+                    rotation_axis=None,
+                    planar_vectors=N.x)  # Positive direction
+    assert J.rotation_axis == N.z
+    assert J.planar_vectors == [N.x, N.y]
+    N, A, P, C = _generate_body()
+    J = PlanarJoint('J', P, C, q0, [q1, q2], u0, [u1, u2],
+                    rotation_axis=None,
+                    planar_vectors=-N.x)  # Negative direction
+    assert J.rotation_axis == -N.z
+    assert J.planar_vectors == [-N.x, N.y]
+    # Test no rotation axis single planar vector nonparallel to x
+    N, A, P, C = _generate_body()
+    J = PlanarJoint('J', P, C, q0, [q1, q2], u0, [u1, u2],
+                    rotation_axis=None,
+                    planar_vectors=[N.x + N.y])
+    assert J.rotation_axis == N.z
+    assert J.planar_vectors == [N.x + N.y, -N.x + N.y]
+    assert C.masscenter.pos_from(
+        P.masscenter) == (q1 - q2) / sqrt(2) * N.x + (q1 + q2) / sqrt(2) * N.y
+    # Test no rotation axis two planar vectors
+    N, A, P, C = _generate_body()
+    J = PlanarJoint('J', P, C, q0, [q1, q2], u0, [u1, u2],
+                    rotation_axis=None,
+                    planar_vectors=[N.x, N.y + N.z])
+    assert J.rotation_axis == -N.y + N.z
+    assert J.planar_vectors == [N.x, N.y + N.z]
+    assert C.masscenter.pos_from(
+        P.masscenter) == q1 * N.x + q2 / sqrt(2) * N.y + q2 / sqrt(2) * N.z
+    # Test rotation axis parallel to y no planar vectors
+    N, A, P, C = _generate_body()
+    J = PlanarJoint('J', P, C, q0, [q1, q2], u0, [u1, u2],
+                    rotation_axis=N.y,  # Positive direction
+                    planar_vectors=None)
+    assert J.rotation_axis == N.y
+    assert J.planar_vectors == [N.x, -N.z]
+    N, A, P, C = _generate_body()
+    J = PlanarJoint('J', P, C, q0, [q1, q2], u0, [u1, u2],
+                    rotation_axis=-N.y,  # Negative direction
+                    planar_vectors=None)
+    assert J.rotation_axis == -N.y
+    assert J.planar_vectors == [-N.x, -N.z]
+
+    # Test error messages
+    N, A, P, C = _generate_body()
+    plan = lambda rotation_axis, planar_vectors: PlanarJoint(
+        'J', P, C, q0, [q1, q2], u0, [u1, u2], rotation_axis=rotation_axis,
+        planar_vectors=planar_vectors)
+    # Test too many planar vectors
+    raises(ValueError, lambda: plan(None, [N.x, N.y, N.z]))
+    # Test wrong planar vector type
+    raises(TypeError, lambda: plan(None, N.x.to_matrix(N)))
+    # Test parallel planar vectors
```

Comment:
How confident are you that the `trigsimp` approach suffices for all reasonable instances of parallel vectors?

Reply:
Not really, but if someone manages to make it really complex, that `trigsimp` does not manage to simplify it. I'd rather throw an error, then keep trying to simplify it forever. You could argue more about the fact whether my checks whether something is parallel with an axis, which don't use a simplify, should get some kind of simplify wrapper.